test/integration/provider/gcp.go

```diff
@@ -0,0 +1,200 @@
+	var machines []string
+	var sPI providerDriver.PluginSPIImpl
+	driverprovider := providerDriver.NewGCPPlugin(&sPI)
+	machineList, err := driverprovider.ListMachines(context.TODO(), &driver.ListMachinesRequest{
+		MachineClass: machineClass,
+		Secret:       &v1.Secret{Data: secretData},
+	})
+	if err != nil {
+		return nil, err
+	} else if len(machineList.MachineList) != 0 {
+		fmt.Printf("\nAvailable Machines: ")
+		for _, machine := range machineList.MachineList {
+			fmt.Printf("%s\n", machine)
+			machines = append(machines, machine)
+		}
+	}
+	return machines, nil
+}
+
+// GetInstancesWithTag describes the instance with the specified tag
+func GetInstancesWithTag(ctx context.Context, svc *compute.Service, searchTagName string, machineClass *v1alpha1.MachineClass, secretData map[string][]byte) ([]string, error) {
+	var instancesID []string
+	project, err := providerDriver.ExtractProject(secretData)
+	if err != nil {
+		return nil, err
+	}
+
+	var providerSpec *api.GCPProviderSpec
+
+	err = json.Unmarshal([]byte(machineClass.ProviderSpec.Raw), providerSpec)
+	if err != nil {
+		providerSpec = nil
+		log.Printf("Error occured while performing unmarshal %s", err.Error())
+		return instancesID, err
+	}
+
+	zone := providerSpec.Zone
+	req := svc.Instances.List(project, zone)
+	if err := req.Pages(ctx, func(page *compute.InstanceList) error {
+		for _, server := range page.Items {
+			//in gcp the tags are just string, not key value pair
+			for _, tag := range server.Tags.Items {
+				if tag == searchTagName {
+					instancesID = append(instancesID, server.Name)
+					break
+				}
+			}
+		}
+		return nil
+	}); err != nil {
+		return instancesID, err
+	}
+
+	return instancesID, nil
+}
+
+// GetAvailableVolumes describes volumes with the specified tag
+func GetAvailableVolumes(ctx context.Context, svc *compute.Service, tagName string, tagValue string, machineClass *v1alpha1.MachineClass, secretData map[string][]byte) ([]string, error) {
+	var availVolID []string
+	project, err := providerDriver.ExtractProject(secretData)
+	if err != nil {
+		return nil, err
+	}
+
+	var providerSpec *api.GCPProviderSpec
+
+	err = json.Unmarshal([]byte(machineClass.ProviderSpec.Raw), providerSpec)
+	if err != nil {
+		providerSpec = nil
+		log.Printf("Error occured while performing unmarshal %s", err.Error())
+		return availVolID, err
+	}
+
+	zone := providerSpec.Zone
+	req := svc.Disks.List(project, zone)
+
+	//check whether these tags are present on the disks or not
+	if err := req.Pages(ctx, func(page *compute.DiskList) error {
+		for _, disk := range page.Items {
+			for key, value := range disk.Labels {
+
+				// #the disks didn't have any tag, check after getting access
+				if strings.Contains(key, tagName) && strings.Contains(value, tagValue) {
+					fmt.Printf("%s", disk.Name)
+					availVolID = append(availVolID, disk.Name)
+
+					//delete the disk
+					DeleteVolume(ctx, svc, project, zone, disk.Name)
+					break
+				}
+			}
```

Comment:
This can be better coded as below. Please verify the logic once again & let me know.
```suggestion
			for value, ok := disk.Labels[tagName]; ok && value == tagValue  {

					fmt.Printf("%s", disk.Name)
					availVolID = append(availVolID, disk.Name)

					//delete the disk
					DeleteVolume(ctx, svc, project, zone, disk.Name)
					break
	
			}
```